### PR TITLE
fix: enable support for TLS 1.2

### PR DESCRIPTION
**Pull request by iamKunalGupta**
- fix for usage with confluent cloud

### CODE REVIEW

**Review by iskakaushik — APPROVED** [review]

File: flow/connectors/kafka/kafka.go [hunk 1]
```diff
@@ -45,7 +47,7 @@ func NewKafkaConnector(
 		kgo.SoftwareNameAndVersion("peerdb", peerdbenv.PeerDBVersionShaShort()),
 	)
 	if !config.DisableTls {
-		optionalOpts = append(optionalOpts, kgo.DialTLSConfig(&tls.Config{MinVersion: tls.VersionTLS13}))
+		optionalOpts = append(optionalOpts, kgo.DialTLSConfig(&tls.Config{MinVersion: tls.VersionTLS12}))
```
Comment (serprex) [inline flow/connectors/kafka/kafka.go:50]: This should be a setting. FedRAMP requires 1.3

File: flow/connectors/kafka/kafka.go [hunk 2]
```diff
@@ -10,6 +10,8 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/twmb/franz-go/plugin/kslog"
+
```
Comment (serprex) [inline flow/connectors/kafka/kafka.go:14]:
```suggestion
```

File: flow/connectors/kafka/kafka.go (hunk 1 above)
Comment (iamKunalGupta) [reply flow/connectors/kafka/kafka.go:50]: But this is minimum supported, max supported is still 1.3

Comment (iamKunalGupta) [reply flow/connectors/kafka/kafka.go:48]: https://github.com/golang/go/blob/go1.22.2/src/crypto/tls/common.go#L738-L751

Comment (serprex) [reply flow/connectors/kafka/kafka.go:48]: Yes. Guess we can always switch to an environment variable at some point to block TLS 1.2 if a customer wants that

**Review by serprex — APPROVED** [review]

**Review by iskakaushik — APPROVED** [review]

**Pull request merged**